src/main/java/com/google/step/youtube/DescriptionParser.java

```diff
@@ -52,14 +54,34 @@ public Pattern getPattern() {
      * @param description of the YouTube video to be parsed.
      * @return A list of all matches found in description.
      */
-    public static List<String> findMatches(Pattern pattern, String description) {
-        List<String> matches = new ArrayList<>();
+    public static List<OfferSnippet> findMatches(Pattern pattern, String description) {
+        List<OfferSnippet> matches = new ArrayList<>();
         Matcher matcher = pattern.matcher(description);
 
         while (matcher.find()) {
-            matches.add(matcher.group());
+            matches.add(OfferSnippet.create(matcher.group(), getSnippet(matcher.start(), description)));
         }
         return matches;
     }
 
+    // Finds the snippet (line) of description which conatins the promocode index. 
+    // Bounds the snippet at MAX_SNIPPET_LENGTH characters and does not truncate words.
+    private static String getSnippet(int promocodeIndex, String description) {
+        String delineator = "\n";
+        int startDelineator = description.lastIndexOf(delineator, promocodeIndex);
+        int endDelineator = description.indexOf(delineator, promocodeIndex);
+
+        int startSnippet = startDelineator == -1 ? 0 : startDelineator + delineator.length();
+        int endSnippet = endDelineator == -1 ? description.length() : endDelineator;
+        String completeSnippet = description.substring(startSnippet, endSnippet);
+
+        if (completeSnippet.length() <= MAX_SNIPPET_LENGTH) {
+            return completeSnippet;
+        }
+
+        int startIndexByWord = description.indexOf(" ", promocodeIndex - MAX_SNIPPET_LENGTH/2) + 1;
```

Comment:
optiona/nit: whitespace isn't exactly to the java style guide.  i dont' care just noticed.

Reply:
Changed to adhere to style guide.